internal/controller/drpolicy_peerclass.go

```diff
@@ -60,19 +64,22 @@ type peerInfo struct {
 	offloaded bool
 }
 
-// peerClassMatchesPeer compares the storage class name across the PeerClass and passed in peer for a match, and if
-// matched compares the clusterIDs that this peer represents. No further matching is required to determine a unique
-// PeerClass matching a peer
+// peerClassMatchesPeer compares the GroupReplicationID (if available) or storage class name between the PeerClass
+// and the passed in peer for a match, and if matched, proceeds to compare the clusterIDs that this peer represents.
+// Prefers GroupReplicationID matching when available, or else use StorageClassName.
+// No further matching is required to determine a unique PeerClass matching a peer.
 func peerClassMatchesPeer(pc ramen.PeerClass, peer peerInfo) bool {
-	if pc.StorageClassName != peer.storageClassName {
-		return false
-	}
-
-	if !equalClusterIDSlices(pc.ClusterIDs, peer.clusterIDs) {
-		return false
+	var match bool
+	// use grid as preference
+	if pc.GroupReplicationID != "" && peer.groupReplicationID != "" {
+		match = pc.GroupReplicationID == peer.groupReplicationID &&
+			pc.StorageClassName == peer.storageClassName
+	} else {
+		// use storageClass when grid is not available
+		match = pc.StorageClassName == peer.storageClassName
 	}
```

Comment:
I think we can just remove all this and just use `match = pc.GroupReplicationID == peer.groupReplicationID` i.e use only storageClass matching.